### PR TITLE
[WIP] Add mAP metric

### PR DESCRIPTION
**Pull request by fsx950223**
I implemented mAP with keras metric. But it can only execute under eager function enabled context. It's a little hard to run it with graph mode. Could you give me a hand?@Squadrick

### CODE REVIEW

**Comment by fsx950223** [comment]
It's so difficult to implement it by Metric

**Comment by Squadrick** [comment]
@fsx950223 Yeah, sure. I'll take a look at this.

**Comment by fsx950223** [comment]
> @fsx950223 Yeah, sure. I'll take a look at this.

Thanks

File: tensorflow_addons/metrics/map.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    def iou(self, b1, b2):
+        """
+        Args
+            b1: bbox.
+            b1: the other bbox.
+        Returns:
+            iou float `Tensor`.
+        """
+        b1_ymin = tf.minimum(b1[:, 0], b1[:, 2])
```
Comment (lgeiger) [inline tensorflow_addons/metrics/map.py]: It looks like this line fails with when I try to run the unittests
```
InvalidArgumentError: slice index 0 of dimension 1 out of bounds. [Op:StridedSlice] name: map/while/strided_slice/
```

Comment (fsx950223) [reply tensorflow_addons/metrics/map.py]: I will fix it later

Comment (lgeiger) [reply tensorflow_addons/metrics/map.py]: Thanks for taking a look

Comment (lgeiger) [reply tensorflow_addons/metrics/map.py]: Any updates on this?

Comment (fsx950223) [reply tensorflow_addons/metrics/map.py]: Here is my [implementation](https://github.com/fsx950223/mobilenetv2-yolov3/blob/d8ecb19b8ffd6f9f2bde6d313de609eebd44614a/yolo3/map.py#L120) of map.You can also revert to first commit to getting the runnable version.

**Comment by gabrieldemarmiesse** [comment]
I'll close this pull request as it has been inactive for a while now. Feel free to comment if you want me to reopen the pull request.

**Pull request closed without merging**